GroceryList/app/src/main/java/com/mtruehle/grocerylist/MainActivity.java

```diff
@@ -0,0 +1,152 @@
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity {
+    private ArrayList<String> itemsToBuy;
+    private ArrayAdapter<String> itemsAdapter; // referred to by class methods.
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        // starts up the main activity, defines necessary parameters like itemsToBuy, starts up the ListView, and adds Lucky Charms to make testing stuff out slightly quicker.
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        itemsToBuy = new ArrayList<String>();
+        itemsToBuy.add("Lucky Charms"); // they're magically delicious.
+        itemsAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, itemsToBuy);
+        ListView itemListView = (ListView) findViewById(R.id.groceryList);
+        itemListView.setAdapter(itemsAdapter);
+        itemListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView parent, View view, int position, long id) {
+                handleListClick(view, position); // separating the dialog and handling for clicking, mainly for ease of reading and debugging.
+            }
+        });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.menu_main, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.delete_all:
+                AlertDialog.Builder areYouSure = new AlertDialog.Builder(this);
+                areYouSure.setTitle(R.string.are_you_sure);
+                areYouSure.setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        itemsToBuy.clear();
+                        itemsAdapter.notifyDataSetChanged();
+                        Toast.makeText(MainActivity.this, R.string.cleared, Toast.LENGTH_SHORT).show();
+                    }
+                });
+                areYouSure.setNegativeButton(R.string.cancel_button, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        // toast here? or is that too much feedback?
```

Comment:
I think you're call was good, later we are going to start to get into how you could work with users to figure out if that is too much feedback or not.